services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeController.kt

```diff
@@ -92,13 +92,15 @@ class HankeController(
     /** Add one hanke. This method will be called when we do not have id for hanke yet */
     @PostMapping
     fun createHanke(@ValidHanke @RequestBody hanke: Hanke?): Hanke {
-        val userId = currentUserId()
-        logger.info { "Creating Hanke for user $userId: ${hanke?.toLogString()} " }
-
         if (hanke == null) {
             throw HankeArgumentException("No hanke given when creating hanke")
         }
-        val createdHanke = hankeService.createHanke(hanke)
+        val sanitizedHanke = hanke.copy(id = null, generated = false)
+
+        val userId = currentUserId()
+        logger.info { "Creating Hanke for user $userId: ${hanke.toLogString()} " }
+
+        val createdHanke = hankeService.createHanke(sanitizedHanke)
```

Comment:
I think deciding what to do with the generated value is business logic so it should be done in the service. There are other fields that are reset when creating a hanke, so this could be added there.
https://github.com/City-of-Helsinki/haitaton-backend/blob/59091c289426cdc6e135f240a131b02b998c3920/services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt#L124-L132

Reply:
createHanke method is used both for generated and non-generated hanke. How would we know what value to set for generated? In my opinion sanitising  user input is not necessarily business logic. I see that as a safety measure so that nothing funky ends up in the service. I guess there could be a specific mapper to handle transformation from user input -> actual data to be handled.

Reply:
Yes, I forgot the method is called from both flows. Once again the endpoint-specific DTOs would save us from this headache. I guess this is fine as-is.